Tests/DatadogTests/Datadog/RUM/RUMVitals/VitalCPUReaderTest.swift

```diff
@@ -0,0 +1,64 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+import UIKit.UIApplication
+@testable import Datadog
+
+class VitalCPUReaderTest: XCTestCase {
+    let testNotificationCenter = NotificationCenter()
+    lazy var cpuReader = VitalCPUReader(notificationCenter: testNotificationCenter)
+
+    func test_whenCPUUnderHeavyLoad_itMeasuresHigherCPUTicks() throws {
+        let highLoadAverage = try averageCPUTicks {
+            for _ in 0...500_000 {
+                let random = Double.random(in: Double.leastNonzeroMagnitude...Double.greatestFiniteMagnitude)
+                _ = tan(random).squareRoot()
+            }
+        }
+
+        let lowLoadAverage = try averageCPUTicks {
+            Thread.sleep(forTimeInterval: 1.0)
+        }
+
+        XCTAssertGreaterThan(highLoadAverage, lowLoadAverage)
```

Comment:
This test seems stable when running on a Mac, but when running on an iPhone, I receive ~13% failures. And it's coming from a reason that the `highLoadAverage` is sometimes `0`. While we can call the last 3 failures a "flakiness" factor, the first 10 do rather look like some measurement issue, WDYT?

<img width="1393" alt="Screenshot 2021-06-11 at 08 39 30" src="https://user-images.githubusercontent.com/2358722/121642467-dace8100-ca90-11eb-931f-e6c2049cc1b5.png">

Reply:
now i'm running `heavyLoad()` for 1.0 second. that should improve test results.

although usually `host_statistics` updates its values after 0.01 second (which makes sense when we look at its source code), it sometimes keeps reporting the same values for 1.0 second (which doesn't make sense to me).
never 2.0 seconds or 0.5 second. i still can't figure out why.

i just noticed that when i do `Thread.sleep(...)`, `host_statistics` returns updated values after 0.01 second most of the time. 
without `Thread.sleep(...)` it usually doesn't update for 1.0 second.